utils/node_local_test.py

```diff
@@ -3114,6 +3120,438 @@ def test_dfuse_perms(self):
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=dfuse.dir)
+        if rc.stderr != b'':
+            print('Stderr from command')
+            print(rc.stderr.decode('utf-8').strip())
+        if rc.stdout != b'':
+            print(rc.stdout.decode('utf-8').strip())
+
+        dir_list = os.listdir(dir1)
+        nr_entries = len(dir_list)
+        if nr_entries != 3:
+            raise NLTestFail('Wrong number of entries')
+
+        cmd = [daos_mw_fi, self.pool.id(), self.container.id(), "punch_entry", "/test_dir"]
+        print(cmd)
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=dfuse.dir)
+        if rc.stderr != b'':
+            print('Stderr from command')
+            print(rc.stderr.decode('utf-8').strip())
+        if rc.stdout != b'':
+            print(rc.stdout.decode('utf-8').strip())
+        assert rc.returncode == 0
+
+        # run the checker while dfuse is still mounted (should fail - EX open)
+        cmd = ['fs', 'check', self.pool.id(), self.container.id(), '--flags', 'print', '--dir-name',
+               'lf1']
+        rc = run_daos_cmd(self.conf, cmd, log_mask='CRIT')
+        print(rc)
+        assert rc.returncode != 0
+        output = rc.stderr.decode('utf-8')
+        line = output.splitlines()
+        if line[-1] != 'ERROR: daos: failed fs check: errno 16 (Device or resource busy)':
+            raise NLTestFail('daos fs check should fail with EBUSY')
+
+        # stop dfuse
+        if dfuse.stop():
+            self.fatal_errors = True
+
+        # fs check with relink should find the 2 leaked directories.
+        # Everything under them should be relinked but not reported as leaked.
+        cmd = ['fs', 'check', self.pool.id(), self.container.id(), '--flags', 'print,relink',
+               '--dir-name', 'lf1']
+        rc = run_daos_cmd(self.conf, cmd)
+        print(rc)
+        assert rc.returncode == 0
+        output = rc.stdout.decode('utf-8')
+        line = output.splitlines()
+        if line[-1] != 'Number of Leaked OIDs in Namespace = 2':
+            raise NLTestFail('Wrong number of Leaked OIDs')
+
+        # run again to check nothing is detected
+        cmd = ['fs', 'check', self.pool.id(), self.container.id(), '--flags', 'print,relink']
+        rc = run_daos_cmd(self.conf, cmd)
+        print(rc)
+        assert rc.returncode == 0
+        output = rc.stdout.decode('utf-8')
+        line = output.splitlines()
+        if line[-1] != 'Number of Leaked OIDs in Namespace = 0':
+            raise NLTestFail('Wrong number of Leaked OIDs')
+
+        # remount dfuse
+        dfuse = DFuse(self.server,
+                      self.conf,
+                      pool=self.pool.id(),
+                      container=self.container,
+                      caching=False)
+        dfuse.log_mask = 'CRIT'
```

Comment:
What is the need to set log_mask here, are there issues with dfuse or is it to limit run-times?

I remember some discussion about this but I can't recall what the issue wasl.

Reply:
correction, it was a skype discussion.
the test adds some faults and expects ERR messages later on daos commands, but NLT causes the stage to fail if it sees those error messages. so for this test, the logging is set to CRIT (which you had suggested) to avoid the NLT stage to fail, but still verifies the output from the daos tool is an error.

Reply:
Yes, I couldn't remember the details.

I've got a better idea here where we leave debugging enabled which would capture any memory leaks but install a log handler that would ignore error messages just for the specific log files and only if they had the "correct" DF_RC suffix, whatever the code is reporting in these cases.